R/ff_prep_refactored.R

```diff
@@ -96,69 +95,14 @@ ff_prep_refactored <- function(datafolder = NA, country = NA, shape = NA, tiles
   }
 
   first <- TRUE
+  # Process tiles and dates
   ####### load raster data as matrix#########
-  for (tile in tiles) {
-    if (exists("extent", inherits = F)) {
-      rm(extent)
-    }
-
-    files <- allfiles[grep(tile, allfiles)]
-
-    shape <- process_shape_country(shape, shrink, files, borders, verbose)
-
-    for (i in dates) {
-      if (exists("dts", inherits = F)) {
-        rm(dts)
-      }
-
-      if (verbose) {
-        cat(paste("loading tile data from", tile, "for", i, " "))
-      }
-
-      selected_files <- process_date_and_groundtruth(i, files, groundtruth_pattern)
-
-      rasstack <- load_raster_data_by_tile(selected_files, shape, shrink, NA, verbose)
-
-      if (length(tiles) > 1) {
-        groundtruth_raster <- NA
-      } else {
-        groundtruth_result <- handle_groundtruth_raster(selected_files, groundtruth_pattern, first, verbose, extent, hasgroundtruth)
-        groundtruth_raster <- groundtruth_result$groundtruth_raster
-        hasgroundtruth <- groundtruth_result$hasgroundtruth
-      }
-
-      # Process raster data
-      dts <- process_raster_data(rasstack, shape, shrink, addxy, dts, coords)
-
-      # Add date features if necessary
-      if (adddate) {
-        dts <- add_date_features(dts, i)
-      }
-
-      result <- finalize_data(dts, selected_files, addxy, adddate)
-      dts <- result$dts
-      newcolnames <- result$newcolnames
-
-      # filter on filter conditions
-
-      filterresult <- filter_by_feature(fltr_features, fltr_condition, dts, verbose = verbose)
-      dts <- filterresult$filtered_matrix
-      sf_indices <- filterresult$filtered_indices
-
-      # take a random sample if that was applied
+  process_result <- process_tile_data(tiles, allfiles, shape, shrink, window, borders, verbose, dates, groundtruth_pattern, hasgroundtruth, first, addxy, adddate, fdts, sample_size, allindices, fltr_features, fltr_condition)
 
-      # Subset matrices based on common column names
-      # Merge matrices by column names
-      combine_result <- sample_and_combine_data(dts, fdts, sf_indices, sample_size, first, allindices, verbose)
-      fdts <- combine_result$fdts
-      allindices <- combine_result$allindices
-      first <- combine_result$first
-    }
-
-    if (verbose) {
-      cat(paste("loading finished, features:", paste(newcolnames, collapse = ", "), "\n"))
-    }
-  }
+  fdts <- process_result$fdts
```

Comment:
Any clue what fdts means?

Reply:
feature dataset or final dataset

Reply:
That makes sense, might be nice to spell it out, but not necessary I guess

Reply:
yeah, I didn't want to change their variables, since it might be confusing for them, we can suggest them to do so